monty_profiling/floppy/flop_counting/operations/arithmetic.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import warnings
+from typing import Any, Optional, Tuple
+
+__all__ = ["Addition", "Subtraction", "Multiplication", "Division"]
+
+class ArithmeticOperation:
+    """Base class for arithmetic operations."""
+
+    def __init__(self, name: str):
+        self.name = name
+
+    def _compute_broadcast_shape(
+        self, *shapes: Tuple[int, ...]
+    ) -> Optional[Tuple[int, ...]]:
+        """Compute the broadcast shape for the inputs."""
+        try:
+            # Handle case where one input is scalar
+            if not all(shapes):
+                return shapes[1] if not shapes[0] else shapes[0]
+
+            # Create dummy arrays of ones with the given shapes
+            arrays = [np.ones(shape) for shape in shapes]
+            result = np.broadcast_arrays(*arrays)
+            return result[0].shape
+        except ValueError as e:
+            warnings.warn(f"Invalid broadcast shape: {str(e)}")
+            return None
+
+    def count_flops(self, *args: Any, result: Any) -> Optional[int]:
```

Comment:
It would be helpful with more detailed doc strings for key functions like this one, e.g. something like the below:

```python
        """Count the FLOPs (Floating Point Operations) for the operation.

        This function estimates the number of FLOPs required to perform a numpy-like operation. It handles both scalar and array inputs
        and accounts for broadcasting by using the shape of the actual result.

        Args:
            *args (Any): Input arguments for the operation. Can include scalars or arrays.
            result (Any): The result of the operation, used to determine the output shape.

        Returns:
            Optional[int]: The estimated number of FLOPs, or None if the input is invalid.
        """
```